rb/lib/selenium/webdriver/remote/bridge.rb

```diff
@@ -605,11 +552,58 @@ def find_elements_by(how, what, parent = nil)
           ids.map { |id| Element.new self, element_id_from(id) }
         end
 
+        protected
+
+        def capabilities_class
+          W3CCapabilities
+        end
+
+        def default_capabilities
+          capabilities_class.firefox
+        end
+
         private
 
-        def assert_javascript_enabled
-          return if capabilities.javascript_enabled?
-          raise Error::UnsupportedOperationError, 'underlying webdriver instance does not support javascript'
+        def bridge_module
+          Module.nesting[1]
+        end
+
+        def process_capabilities(opts)
+          capabilities = opts.delete(:desired_capabilities) || default_capabilities
+          capabilities = Capabilities.send(capabilities) if capabilities.is_a?(Symbol)
```

Comment:
We merge with defaults right now for firefox (w3c bridge) because someone in #watir was passing in a hash for desired capabilities (without `browser_name`) instead of a `Capabilities` class and apparently now geckodriver needs `browser_name` set.

That being said, I agree it would be a nice option for people to specify capabilities as a hash without merging defaults so nevermind on that one. 

The second comment was an example, not saying `firefox` specifically. I was more referring to the processing of `:options` and `:profile` in the `Capabilities` class itself instead of doing in the `Capabilities` class and then in `process_capabilities` [here](https://github.com/SeleniumHQ/selenium/pull/3490/files/7fee63589f33dab25d3d52bd27f0694c10388a80#diff-dd335728ad314dd52eece6507da94ea5R575) but I don't have a solution for that more just bringing it up as an idea